AppCenter/AppCenter/MSAppCenter.m

```diff
@@ -414,19 +414,30 @@ - (void)setLogUrl:(NSString *)logUrl {
 
 - (void)setMaxStorageSize:(long)sizeInBytes completionHandler:(void (^)(BOOL))
 completionHandler {
-  if (self.configuredFromApplication) {
-    MSLogWarning([MSAppCenter logTag],
-        @"Unable to set storage size after the application has configured App"
-        "Center");
-    if (completionHandler) {
-      completionHandler(NO);
+  @synchronized (self) {
+    if (self.setMaxStorageSizeHasBeenCalled) {
+      MSLogWarning([MSAppCenter logTag], @"setMaxStorageSize:completionHandler: may only be called once per app "
+                                         "launch");
+      if (completionHandler) {
+        completionHandler(NO);
```

Comment:
Also, since we are interacting with the database, I think we should move most of the work on a background thread so that this call doesn't block. (We probably meant to do that but forgot to)

Reply:
Yes I agree, anything that's executing SQL statement should be done in a background thread.